### ISSUE TITLE
Tracking issue for `--extern nounused`

### ISSUE THREAD

**Issue by ehuss** [rust-lang/rust#98400]
This is a tracking issue for the `nounused` CLI option for `--extern`.

* Docs: https://doc.rust-lang.org/nightly/unstable-book/compiler-flags/extern-options.html

This option marks the given crate so that it does not trigger [`unused_crate_dependencies`](https://doc.rust-lang.org/nightly/rustc//lints/listing/allowed-by-default.html#unused-crate-dependencies) warnings for the crate.

This is useful if the given crate *might* be unused, but shouldn't cause a warning.
For example, when using `--extern` to link with standard library crates, and the user does not explicitly use one of those crates.

The syntax is `[opts ':'] name ['=' path]` where opts is a comma separated list of options:

```
--extern nounused:foo=/path/to/libfoo.rlib
```

See also #xxx and #xxx which share this syntax extension.

### About tracking issues

Tracking issues are used to record the overall progress of implementation.
They are also used as hubs connecting to other relevant issues, e.g., bugs or open design questions.
A tracking issue is however *not* meant for large scale discussion, questions, or bug reports about a feature.
Instead, open a dedicated issue for the specific matter and add the relevant feature gate label.

### Steps

- [X] Implementation: #96067
- [ ] Adjust documentation ([see instructions on rustc-dev-guide][doc-guide])
- [ ] Stabilization PR ([see instructions on rustc-dev-guide][stabilization-guide])

[stabilization-guide]: https://rustc-dev-guide.rust-lang.org/stabilization_guide.html#stabilization-pr
[doc-guide]: https://rustc-dev-guide.rust-lang.org/stabilization_guide.html#documentation-prs

### Unresolved Questions

* [ ] The syntax for adding options to `--extern` flags was added without much bikeshedding or discussion. It may need adjusting to be more flexible for more complex options.

### Implementation history

- #96067

**Comment by petrochenkov** [comment]
Stabilization blocked on ["Tracking Issue `--extern` crate modifiers"](https://github.com/rust-lang/rust/issues/98405).

**Comment by est31** [comment]
I think the name is a bit hard to read. What about `no_unused` or `allow_unused`?

**Comment by jsgf** [comment]
I don't care particularly, but it should be consistent with `noprelude`.

Since these aren't at all user-facing, I'm not too concerned about readability so long as they make sense to someone who does have to deal with them.

**Comment by est31** [comment]
I see, the `noprelude` argument convinced me.

**Comment by ehuss** [comment]
#98405 is proposing to possibly use `+`/`-` syntax like `-prelude` or `+priv`.
